common/djangoapps/student/roles.py

```diff
@@ -134,7 +134,7 @@ def has_user(self, user):
         """
         Return whether the supplied django user has access to this role.
         """
-        if not (user.is_authenticated() and user.is_active):
+        if not (getattr(user, 'is_authenticated', True) and user.is_active):
```

Comment:
Hmm, looking [at the docs](https://docs.djangoproject.com/en/1.9/ref/contrib/auth/#django.contrib.auth.models.User.is_authenticated), it seems like this is supposed to be a method, and `getattr(user, 'is_authenticated', True` is bad because it would return `True` for an `AnonymousUser` (which has an `is_authenticated()` method attribute that returns False).

Sorry for the bad advice. Seems like it is better to leave this as it was and to mock with `is_authenticated = lambda: True` in the case where `AuthenticationMiddleware` is not used.